### ISSUE TITLE
Anomaly in unshelve match goal with

### ISSUE THREAD

**Issue by andres-erbsen** [coq/coq#7488]
#### Version

~~~~~
The Coq Proof Assistant, version 8.8.0 (May 2018)
compiled on May 3 2018 7:11:49 with OCaml 4.06.0
~~~~~

#### Operating system

Arch Linux

#### Description of the problem

~~~~~
Goal False.
  lazymatch goal with _ => idtac end.
  match goal with _ => idtac end.
  unshelve lazymatch goal with _ => idtac end.
  (* Error: *)
  (* Anomaly *)
  (* "File "plugins/ltac/tacinterp.ml", line 1054, characters 35-41: Assertion failed." *)
  (* Please report at http://coq.inria.fr/bugs/. *)
  unshelve match goal with _ => idtac end.
  (* Toplevel input, characters 15-55: *)
  (* Error: *)
  (* Anomaly *)
  (* "File "plugins/ltac/tacinterp.ml", line 1054, characters 35-41: Assertion failed." *)
  (* Please report at http://coq.inria.fr/bugs/. *)
Abort.
~~~~~

This is problematic when using `erewrite ( _ : lhs = _)` inside a repeat match, for example https://github.com/mit-plv/cross-crypto/blob/ce28bd55894d7620cd62f7769f56e475ac010b13/src/Minimalistic.v#L2756

**Comment by JasonGross** [comment]
Workaround: Use `unshelve (idtac; match goal with _ => idtac end)`.  It's an interpretation issue; apparently `unshelve` can't handle tactic expressions; you get the same anomaly with `unshelve (let x _ := idtac in idtac).`.

**Comment by JasonGross** [comment]
cc @ppedrot

**Comment by SkySkimmer** [comment]
Duplicate of #2800.

**Comment by SkySkimmer** [comment]
Nevermind the examples in #2800 pass now but this one doesn't.

**Comment by maximedenes** [comment]
@ppedrot what's your take on this?

**Comment by ppedrot** [comment]
This seems solved on master (13e8d04b2f080fbc7ca169bc39e53c8dd091d279).
